src/main/scala/Checkout.scala

```diff
@@ -1,129 +1,161 @@
+import scala.annotation.tailrec
+
 /**
   * An app to model a supermarket checkout
+  * Source code
   *
   * @author Keoni D'Souza
   */
-
 object Inventory {
 
-  // Char SKUs mapping to prices
-  val stock: Map[Char, Int] = Map[Char, Int](
-    // SKU -> price
-    'A' -> 50,
-    'B' -> 30,
-    'C' -> 20,
-    'D' -> 15
-  )
+  final case class Item(sku: String, price: Int, offer: Option[Offer])
+
+  final case class Offer(quantity: Int, price: Int)
 
-  // Char SKUs mapping to offers
-  val offers: Map[Char, (Int, Int)] = Map[Char, (Int, Int)](
-    // SKU -> (x for y)
-    'A' -> (3, 130),
-    'B' -> (2, 45)
-  )
+  val itemA = Item("A", 50, Option(Offer(3, 130)))
+  val itemB = Item("B", 30, Option(Offer(2, 45)))
+  val itemC = Item("C", 20, None)
+  val itemD = Item("D", 15, None)
 
-  val exampleShop = List('A', 'B', 'C', 'D', 'A', 'A', 'B', 'D', 'D', 'D')
+  val testShop: List[Item] =
+    List(itemA, itemB, itemC, itemD, itemA, itemA, itemB, itemD, itemD, itemD)
 
 }
 
 object Checkout extends App {
 
   import Inventory._
 
+  print(Console.UNDERLINED)
   println("Checkout Kata\n")
+  print(Console.RESET)
 
   /**
     * Calculates subtotal with a recursive definition
     *
     * @param items Products to be purchased
-    * @param stock Char SKUs mapping to prices
-    * @return Item subtotal
+    * @return item subtotal
     */
-  def calculateSubtotalWithRecursion(items: List[Char], stock: Map[Char, Int]): Int =
+  def calculateSubtotalWithRecursion(items: List[Item]): Int =
```

Comment:
add the '@tailrec' annotation

Reply:
I'm not sure that this is tail recursion (only because IntelliJ is telling me it isn't). For example, compare these two:
![image](https://user-images.githubusercontent.com/41479510/53345970-f6d4a580-390d-11e9-930f-732449873d17.png)
![image](https://user-images.githubusercontent.com/41479510/53346072-300d1580-390e-11e9-8956-a15284d22e66.png)

Reply:
And if I add it, it complains:
![image](https://user-images.githubusercontent.com/41479510/53346127-55018880-390e-11e9-8455-a19afccde20a.png)

Reply:
Yeah I just had a proper look and your right its not tail rec as its not returning the function at the end its returning an Int. This is an example of tail rec:

```
import scala.annotation.tailrec
@tailrec
def calculateSubtotalWithRecursion(items: List[Item], price: Int = 0): Int =
  items match {
    case Nil => price
    case _ =>
    calculateSubtotalWithRecursion(items.tail, items.head.price + price)
  }
```

Reply:
Nice. I included an example a couple of lines below:
![image](https://user-images.githubusercontent.com/41479510/53356738-0232cb80-3924-11e9-8ac8-2d4ab1c28359.png)

Reply:
I wrapped it in another function, too, so that you don't need to add a second accumulator parameter when calling it.